File: problems/daily/Java/234. Palindrome Linked List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public boolean isPalindrome(ListNode head) {
        ListNode check = null;
        while(head != null){
            ListNode temp = new ListNode(head.val);
            temp.next = check;
            check = temp;
            head = head.next;
        }
        return check == head;
    }
}

File: problems/daily/Python3/234. Palindrome Linked List.py
    # Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        sl = fs = head

        while fs and fs.next:
            sl_pr = sl
            sl = sl.next
            fs = fs.next.next

        prev = None
        cur = sl

        while cur:
            tent = cur.next
            cur.next = prev
            prev = cur
            cur = tent

        fs_cur = head
        sec_cur = prev

        while fs_cur and sec_cur:
            if fs_cur.val != sec_cur.val:
                return False
            fs_cur = fs_cur.next

            sec_cur = sec_cur.next

        return True